### ISSUE TITLE
Use public enums or static symbols in remaining places where strings are hard-coded

### ISSUE THREAD

**Issue by towerofnix** [leopard-js/leopard#169]
We've defined a few unions where hard-coded string values are used as mode options:

```
export type SpeechBubble = "say" | "think";
public touching(target: "mouse" | "edge" | Sprite | Stage ...)
type WatcherStyle = "normal" | "large" | "slider";
```

However in Trigger.ts, we use static key-symbol properties:

```
public static readonly GREEN_FLAG = Symbol("GREEN_FLAG");
public static readonly KEY_PRESSED = Symbol("KEY_PRESSED");
public static readonly BROADCAST = Symbol("BROADCAST");
...
```

We should consistently go one way or the other. I figure the latter, with key/symbols, is the better way, since even though both are equally arbitrary things to remember, property names tend to stick better; symbols also force you to access by property name vs. an enum where you can "cheat" and just use the enum value. (If my understanding of enums is right!)

Regardless what we decide exactly, since this is a 100% public-facing change, we have to think about how it will look and feel in Leopard projects generated from sb-edit and working on your own Leopard project (from scratch or, ahem, from Scratch).

**Comment by PullJosh** [comment]
I agree that we should pick one and be consistent.

I'll take the controversial stance again and say that I actually prefer opting for using simpler/fewer JS constructs, since we're mostly intending for the library to be used by JavaScript beginners, and so I think I prefer using strings.

Although, when it comes to triggers, maybe we just prefer something like
```js
this.triggers = [
  Trigger.greenFlag(this.whenGreenFlagClicked),
  Trigger.broadcast("message1", this.whenIReceiveMessage1),
  Trigger.keyPressed("space", this.whenKeySpacePressed),
  Trigger.loudnessGreaterThan(10, this.whenGreaterThan)
];
```
I'm not sure why we (I?) even made it so complicated in the first place.

**Comment by towerofnix** [comment]
NB: We don't actually need to stress about `SpeechBubbleStyle` ("say", "think") - it's internal only. It's exposed via `say` and `think` methods, etc.

If we set up triggers to expose through methods too, the only exceptional case is for `touching`: `this.touching("mouse")`, `this.toucing("edge")`. I think it would be OK to break this behavior up into three functions:

```js
this.touchingMouse()
this.touchingEdge()
this.touchingSprite(this.sprites["Sprite2"]) // or whatever
```

(Double NB: I need to verify but Scratch behavior for "touching [sprite name]" is equivalent to `this.touchingSprite(this.sprites[spriteName].andClones)`. I don't think we're reflecting that in Leopard or toLeopard yet. That's a separate issue, of course!)

**Comment by towerofnix** [comment]
Yeah, I think type-safeness is totally doable in generated code — and has real benefits. My only concern is if tsc-generated errors will be effectively legible to users. We might not be able to do much about that without building a completely custom Leopard IDE, which, while potentially awesome, is totally out of scope for now!

If we want to expose TypeScript annotations in CodeSandbox, we'll probably have to work out generating a compatible webpack project. Even if we decide we only want to expose users to pure JavaScript on codesandbox.com (i.e. through leopardjs.com), it'd probably still be a good idea to implement a mode for export which includes a TypeScript and webpack setup - people developing a project with Leopard and TypeScript offline should still have a convenient way to package their project as standalone (and that's not really a skill that is relevant to what Leopard tries to teach).

We don't have a proper offline export built-in yet at all though, so that's still a ways off. IMO it's worth investigating the UI for that before exporting TypeScript from the website.

**Comment by towerofnix** [comment]
Another case of existing enums which should probably be changed:

```js
Sprite.RotationStyle = Object.freeze({
  ALL_AROUND: Symbol("ALL_AROUND"),
  LEFT_RIGHT: Symbol("LEFT_RIGHT"),
  DONT_ROTATE: Symbol("DONT_ROTATE")
});
```

More cases which are currently lowercase strings: audio effect names, visual effect names.

The `touching` function also accepts `Color` inputs, so should be expanded to `touchingColor` as well. (Something to consider is `colorTouching` which is for "does color X on my costume touch color Y on some other sprite's costume / pen trails?" It's named similarly and we may want to rename it — `colorTouchingColor` would be a simple change that delineates it from other "touching" blocks more clearly.)

Those are the only new special cases I noted going over `toLeopard.ts` - I checked pretty carefully but I might have missed some.

**Comment by towerofnix** [comment]
I'm going to go ahead with the changes discussed here so far, we can discuss individual naming decisions and implementation details in a pull request.
